### ISSUE TITLE
hyperparameter evolution question

### ISSUE THREAD

**Issue by firsakov** [ultralytics/yolov5#6345]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi,

I'm interested in the principle of hyperparameter evolution. Please, tell me if I understand correctly.
We first initialize 3 options for all of hyperparameters in train.py. Then train 10 epochs, and choose 2 of them depending on fintess metric. Then we perform crossover (just swap N hyperparameters between two sets), and then mutation (flip of M hyperparameters). 

Parameters for crossover and mutation are chosen randomly? Is different importance of different hyperparameters taken into account? I guess initial lr or weight decay affect more than for example anchors per output grid or image shear. Also, as I understand, algorithm doesn't consider any theory/rules of hyperparameter tuning like decreasing warmup iters if loss isn't going down or increasing learning rate if ...? 

Also question about number of epochs. From my experience, if a set of hyperparameters does well on first 10 epochs, it doesn't mean it will do so on full training.

Thank you in advance.

**Comment by glenn-jocher** [comment]
@firsakov no, evolution starts with the hyps in your train.py --hyp file constrained to the limits in the `meta` dict in train.py. We use a population size of 1 per generation with elitism.

For theory see https://en.wikipedia.org/wiki/Genetic_algorithm

For my implementation in practice see train.py https://github.com/ultralytics/yolov5/blob/e5219099cd6e76e4f75a4c8b376531af2791d358/train.py#L538-L624
